scripts/filename-checker

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+#
+# @author muskater
+# @brief Checks that only files exist that conform to the naming convention: [/.a-zA-Z0-9_-]*
+# @date 18.11.2021
+# @tags validation
+
+if [ $# -eq 0 ]; then
+	echo "Usage: filename-checker <path/to/file>"
+	exit
+fi
+
+SCRIPT_DIR="$(dirname "$0")"
+
+ALLOWLIST=$(mktemp)
+grep -Ev '^(#|[[:space:]]*$)' "$SCRIPT_DIR/../tests/filename.whitelist" > "$ALLOWLIST"
```

Comment:
The script needs to fail if the file couldn't be found. The check only checks for spaces, it should be more strict.